unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/Common/SceneCatalogController.cs

```diff
@@ -0,0 +1,374 @@
+using DCL.Configuration;
+using DCL.Helpers;
+using System;
+using System.Collections.Generic;
+
+public interface ISceneCatalogController
+{
+    event Action OnHideCatalogClicked;
+    event Action<CatalogItem> OnCatalogItemSelected;
+    event Action OnResumeInput;
+    event Action OnStopInput;
+
+    void Initialize(ISceneCatalogView view, IQuickBarController quickBarController);
+    void Dispose();
+    void AssetsPackFilter(bool isOn);
+    void CategoryFilter(bool isOn);
+    void FavoritesFilter(bool isOn);
+    void ToggleCatalogExpanse();
+    void OnSearchInputChanged(string currentSearchInput);
+    List<Dictionary<string, List<CatalogItem>>> FilterAssets(string nameToFilter);
+    void QuickBarInput(int quickBarSlot);
+    void ShowFavorites();
+    void CatalogItemSelected(CatalogItem catalogItem);
+    void OnCatalogItemPackSelected(CatalogItemPack catalogItemPack);
+    void SceneCatalogBack();
+    bool IsCatalogOpen();
+    void ShowCategories();
+    void ShowAssetsPacks();
+    void ShowCatalogContent();
+    void OpenCatalog();
+    void CloseCatalog();
+    void RefreshAssetPack();
+    void RefreshCatalog();
+    CatalogItemAdapter GetLastCatalogItemDragged();
+}
+
+public class SceneCatalogController : ISceneCatalogController
+{
+    public event Action OnHideCatalogClicked;
+    public event Action<CatalogItem> OnCatalogItemSelected;
+    public event Action OnResumeInput;
+    public event Action OnStopInput;
+
+    internal ISceneCatalogView sceneCatalogView;
+    internal IQuickBarController quickBarController;
+    private FavoritesController favoritesController;
+    internal List<Dictionary<string, List<CatalogItem>>> filterObjects = new List<Dictionary<string, List<CatalogItem>>>();
+    internal bool isShowingAssetPacks = false;
+    private bool isFilterByAssetPacks = true;
+    internal const string FAVORITE_NAME = "Favorites";
+
+    public void Initialize(ISceneCatalogView sceneCatalogView, IQuickBarController quickBarController)
+    {
+        this.sceneCatalogView = sceneCatalogView;
+        this.quickBarController = quickBarController;
+        favoritesController = new FavoritesController(sceneCatalogView.catalogGroupList);
```

Comment:
shouldn't `FavoritesController` have a `Dispose` to unsubscribe from `catalogGroupListView.OnCatalogItemFavorite`?

Reply:
You're right, I have added the `Dispose()` method. Thanks! Changes applied in https://github.com/decentraland/explorer/pull/2103/commits/3c8f53b158e801207062591839f40aa6e4a08a7d